app/controllers/storage_manager_controller.rb

```diff
@@ -387,7 +385,9 @@ def set_record_vars(sm, mode = nil)
     # sm.agent_type = @edit[:new][:agent_typ] && @edit[:new][:agent_typ] != "" ? @edit[:new][:agent_typ] : "SMIS"
     sm.zone = Zone.find_by_name(@edit[:new][:zone])
 
-    sm.update_authentication({:default => {:userid => @edit[:new][:userid], :password => @edit[:new][:password]}}, :save => (mode != :validate))
+    sm.update_authentication({:default => {:userid   => @edit[:new][:userid],
+                                           :password => @edit[:new][:password] ? @edit[:new][:password] : sm.authentication_password}},
```

Comment:
@Fryguy Thanks for pointing that out. We should leave ```set_record_vars``` as is.
The original password value will be restored back in the new method ```restore_password``` (when ```"Cancel password change"``` is clicked). In fact, that is how I have implemented this in the other controllers.